### DISCUSSION TITLE
How does Evo compare to using an SPA framework like React?

### DISCUSSION THREAD

**Discussion by JBaczuk** [EvolvedWeb/evowc#31] [Q&A] [answered]
I was just starting a project from bare-bones html and hoping I could simply reuse html between files without using a framework like React (using only browser standards).

**For example:** In one file I have the header as a web component, and another the footer (also WC), then in all my pages i can just reuse them. But, since I had them each in their own files, the front end has to request them separately, which makes sense. So the solution would be to bundle the component markup and their javascript. I was just exploring how close I can get to the developer efficiency of React without the bloat.

**Comment by intervalia** [comment, marked as answer]
Evo can handle both SPAs and MPAs. But it may be a mindset change.

Evo has a router to handle SPA webapps.
Look at the file: [evowc/static/slideshow.html](static/slideshow.html). This shows how the router works.

[`<wc-router>`](components/WcRouterElement.html) and [`<wc-a>`](components/WcAElement.html) are the two components you will want to copy into your project. As well as the files [EvoRouter.js](static/js/EvoRouter.js) and [EvoElement.js](static/js/EvoElement.js). That is the base class that is needed for all the components.

Then dig into the various components I have in the [evowc/components](components) folder.

You can always use a bundler, like [webpack](https://webpack.js.org/), to package all of your JS code together, or you can just import the files you need.

`<wc-router>` allows you to use preloaded the component for that route or you can load them on-demand.

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>EVO slideshow</title>
</head>
<body>
  <wc-router>
    <!-- The file "HomePage.js" is pre-loaded in the script tag below -->
    <wc-route href="/home" element="home-page"></wc-route>
    <!-- Load the file "WelcomePage.js" only when that route is active -->
    <wc-route href="/welcome" element="welcome-page" import="./WelcomePage.js"></wc-route>
  </wc-router>
  <script type="module">
    import "./js/components/WcRouterElement.js";
    import "./js/components/HomePage.js";
  </script>
  <style>
  ... Your page and non shadow DOM styles here ...
  </style>
</body>
</html>
```
